CMSIS/NN/Source/NNSupportFunctions/arm_nn_vec_mat_mult_t_s16.c

```diff
@@ -56,7 +55,114 @@ arm_status arm_nn_vec_mat_mult_t_s16(const q15_t *lhs,
                                      const int32_t activation_min,
                                      const int32_t activation_max)
 {
-    int32_t row_loop_cnt = rhs_rows / 2;
+    const int32_t row_loop_cnt = rhs_rows / 2;
+
+#if defined(ARM_MATH_DSP) && !defined(ARM_MATH_MVEI)
+
+    for (int32_t i = 0; i < row_loop_cnt; i++)
+    {
+        q63_t acc_64_0 = 0;
+        q63_t acc_64_1 = 0;
+        int32_t acc_0 = 0;
+        int32_t acc_1 = 0;
+
+        const int32_t col_loop_cnt = rhs_cols / 4;
+
+        const int16_t *lhs_vec = lhs;
+        const int8_t *rhs_0 = rhs;
+        const int8_t *rhs_1 = rhs + rhs_cols;
+        rhs += 2 * rhs_cols;
+
+        for (int j = col_loop_cnt; j != 0; j--)
+        {
+            int32_t ker_0, ker_1, vec_part_0, vec_part_1;
+            vec_part_0 = arm_nn_read_q15x2_ia(&lhs_vec);
+            vec_part_1 = arm_nn_read_q15x2_ia(&lhs_vec);
+
+            rhs_0 = read_and_pad(rhs_0, &ker_0, &ker_1);
```

Comment:
Looks like read_and_pad() is not adapted to the new intrinsic that we have added.. __SXTB16_RORn

Could you replace SXTB(__ROR()) by __SXTB16_RORn instead so that we get similar performance on GCC as well..?

Reply:
Done